src/message/cose.ts

```diff
@@ -39,7 +39,9 @@ export class CoseMessage {
 
   static fromCborData(data: CborData): CoseMessage {
     const cose = cbor.decodeFirstSync(data, decoders).value
-    const protectedHeader = cbor.decodeFirstSync(cose[0])
+    const protectedHeader = cose[0]?.length
+      ? cbor.decodeFirstSync(cose[0])
+      : cose[0]
```

Comment:
The encoded request we use in tests is anonymous so it has an empty protected header and will throw an error when attempting to decode it. We typically only create messages from CBOR during the parsing of a _response_ (which are always signed) and not a request so we never ran into this situation before.